hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -279,24 +289,22 @@ private void populateResultFromNearCache(Collection<?> keys, List<Object> result
         }
     }
 
-    private Map<Object, Long> getNearCacheReservations(Collection<?> nearCacheKeys, Map<Object, Data> keyMap) {
-        Map<Object, Long> reservations = createHashMap(nearCacheKeys.size());
-        for (Object key : nearCacheKeys) {
-            Data keyData = serializeKeys ? (Data) key : keyMap.get(key);
-            long reservationId = tryReserveForUpdate(key, keyData);
+    private void createNearCacheReservations(Collection<?> dataKeys, Map<Object, Long> reservations) {
```

Comment:
The problem is that I get a `Set<? extends K> keys` from the public API, which contains the key set from the user. Then I have this line:
```java
Collection<?> ncKeys = serializeKeys ? dataKeys : new LinkedList<K>(keys);
```
So here we have the wildcard to be able to have either `<Data>` or `<K>` keys. So when I change the signature of `createNearCacheReservations()` to `Collection<Data>`, I have to make an unchecked cast, which produces a warning. So I have to suppress this (either inlined, which is kind of ugly) or for the whole method (which may hide other cast issues).

This is the reason why I didn't do it in the first place.